batchflow/models/eager_torch/layers/resize.py

```diff
@@ -0,0 +1,324 @@
+""" Resizing Torch layers. """
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..utils import get_shape
+
+
+
+class IncreaseDim(nn.Module):
+    """ Increase dimensionality of passed tensor by one. """
+    def __init__(self, dim=1):
+        super().__init__()
+        self.dim = dim
+
+    def forward(self, x):
+        dim = len(get_shape(x)) - 2 + self.dim
+        ones = [1] * dim
+        return x.view(x.size(0), -1, *ones)
+
+
+class ReduceDim(nn.Module):
+    """ Reduce dimensionality of passed tensor by one. """
+    def __init__(self, dim=1):
+        super().__init__()
+        self.dim = dim
+
+    def forward(self, x):
+        dim = max(len(get_shape(x)) - 2 - self.dim, 0)
+        ones = [1] * dim
+        return x.view(x.size(0), -1, *ones)
+
+
+class Reshape(nn.Module):
+    """ Enforce desired shape of tensor. """
+    def __init__(self, shape=None, reshape_to=None):
+        super().__init__()
+        self.shape = shape or reshape_to
+
+    def forward(self, x):
+        return x.view(x.size(0), *self.shape)
+
+
+
+class Interpolate(nn.Module):
+    """ Upsample inputs with a given factor.
+
+    Notes
+    -----
+    This is just a wrapper around ``F.interpolate``.
+
+    For brevity ``mode`` can be specified with the first letter only: 'n', 'l', 'b', 't'.
+
+    All the parameters should the specified as keyword arguments (i.e. with names and values).
+    """
+    MODES = {
+        'n': 'nearest',
+        'l': 'linear',
+        'b': 'bilinear',
+        't': 'trilinear',
```

Comment:
there are more modes in `torch.nn.functional.interpolate`. Does one have to always keep in mind that 'b' stands for 'bilinear' and not for 'bicubic'?

Reply:
Other modes can be passed directly as full strings, if needed